desc/compute/_field.py

```diff
@@ -950,6 +950,76 @@ def compute_B_dot_gradB(
     return data
 
 
+def compute_boozer_magnetic_field(
+    R_lmn,
+    Z_lmn,
+    L_lmn,
+    i_l,
+    Psi,
+    R_transform,
+    Z_transform,
+    L_transform,
+    iota,
+    data=None,
+    **kwargs,
+):
+    """Compute covariant magnetic field components in Boozer coordinates.
+
+    Parameters
+    ----------
+    R_lmn : ndarray
+        Spectral coefficients of R(rho,theta,zeta) -- flux surface R coordinate.
+    Z_lmn : ndarray
+        Spectral coefficients of Z(rho,theta,zeta) -- flux surface Z coordiante.
+    L_lmn : ndarray
+        Spectral coefficients of lambda(rho,theta,zeta) -- poloidal stream function.
+    i_l : ndarray
+        Spectral coefficients of iota(rho) -- rotational transform profile.
+    Psi : float
+        Total toroidal magnetic flux within the last closed flux surface, in Webers.
+    R_transform : Transform
+        Transforms R_lmn coefficients to real space.
+    Z_transform : Transform
+        Transforms Z_lmn coefficients to real space.
+    L_transform : Transform
+        Transforms L_lmn coefficients to real space.
+    iota : Profile
+        Transforms i_l coefficients to real space.
+
+    Returns
+    -------
+    data : dict
+        Dictionary of ndarray, shape(num_rho,) of covariant magnetic field components
+        in Boozer coordinates.
+
+    """
+    grid = R_transform.grid
+    data = compute_covariant_magnetic_field(
+        R_lmn,
+        Z_lmn,
+        L_lmn,
+        i_l,
+        Psi,
+        R_transform,
+        Z_transform,
+        L_transform,
+        iota,
+        data=data,
+    )
+
+    if check_derivs("I", R_transform, Z_transform, L_transform):
+        data["I"] = surface_averages(grid, data["B_theta"])
+    if check_derivs("I_r", R_transform, Z_transform, L_transform):
+        data["I_r"] = surface_averages(grid, data["B_theta_r"])
+    if check_derivs("G", R_transform, Z_transform, L_transform):
+        data["G"] = surface_averages(grid, data["B_zeta"])
```

Comment:
I like our change to compute the currents globally in `compute_boozer_magnetic_field` instead of locally in `compute_qs_error`. But [compute_boozer_coordinates](https://github.com/PlasmaControl/DESC/blob/94bef52844b26ff22cba44f69c356134b73348e2/desc/compute/_qs.py#L99) sets `data["I"]` and `data["G"]` too.

- Do these functions both compute the toroidal and poloidal currents? The one in `compute_boozer_coordinates` doesn't do a surface average of B_theta. (You need a surface average to relate B_theta to I).
- Does  `compute_boozer_coordinates` compute the currents over the whole grid, like we do here?
- I don't like the idea of having two compute functions for one quantity. If they are not the same, one could overwrite the other.

Reply:
That line you linked to is the surface average of B_theta on a surface. That array `B_theta_mn` is the fourier coefficients of `B_theta` on the surface, and the `idx0` index of it is the `m=0,n=0` mode, which is equal to the surface average (since in a surface average of a Fourier Series, every term with `m,n` not equal to zero will integrate to zero, leaving just the constant term)

It currently does NOT compute the currents over the entire grid I believe, only over a single flux surface. (@ddudt the QS objective for boozer harmonics is supposed to only be on a grid at a single given rho value right? may be good to specify that in the docstring as currently I don't think that is written)

I don't think this is a conflict though between the two functions, as the `compute_boozer_coordinates` computes `I` and `G` in a specific way (uses the Fourier Transform of `B_theta`) since it also needs the other harmonics to use in the computation of the Boozer harmonics. So I am ok with having both functions exist/ not making the `compute_boozer_coordinates` function calculate the currents like in  `compute_boozer_magnetic_field`

Reply:
As Dario mentioned, the two methods are equivalent. And there is a test to make sure both functions yield the same results. 

The only difference now is that `compute_boozer_coordinates` can only handle a single flux surface, while `compute_boozer_magnetic_field` can handle arbitrary grids. That's why the later is the default method for computing these currents. 

This could lead to problems with overwriting the values if different grids are used, but that is always an issue. The Boozer coordinates function gets `I` and `G` as a by-product, so the only question is whether to add it to the `data` dict or leave them as local variables. I say keep it as is for now, and we can reassess later.